js/annotations/annotations.src.js

```diff
@@ -1093,6 +1093,59 @@ chartProto.callbacks.push(function (chart) {
         chart.plotBoxClip.destroy();
         chart.controlPointsGroup.destroy();
     });
+    addEvent(chart, 'exportData', function (event) {
+        var annotations = chart.annotations;
+        annotations.forEach(function (annotation) {
+            annotation.labels.forEach(function (label) {
+                var annotationText = label.options.text;
+                label.points.forEach(function (points) {
+                    var annotationX = points.x, xAxisIndex = points.series.xAxis ? points.series.xAxis.options.index : -1;
+                    var wasAdded = false;
+                    if (xAxisIndex === -1) {
+                        var newRow = new Array(event.dataRows[0].length);
```

Comment:
Im curious of using array instead of an object here.

Reply:
It is an array to match the existing format coming from export-data module. `newRow` is later pushed into `dataRows` array that stores arrays - rows of table for data export.

Reply:
Now makes sense, thanks!